app/controllers/messages_controller.rb

```diff
@@ -1,6 +1,7 @@
 class MessagesController < ApplicationController
   skip_before_action :authenticate_user!
-  before_action :set_conversation, only: [:index, :new, :create]
+  before_action :set_current_assistant, only: [:new, :create]
```

Comment:
I made an interesting design decision that I'm now questioning: I save the assistant onto `Current.assistant`. The reason is because assistant_id exists in the URL so it gets passed in as params. But when I get to `def create`, the new message takes `message_params` and assistant_id is not an attribute of message. However, creating a message has the side effect of creating a new conversation (this is within model.rb) and the creation of this new conversation needs assistant_id.

So how to get assistant_id from the controller down into message? I did it through Current.assistant.

The code ends up reasonably clean, but it feels a little odd to put something into Current which will not be available upon *every* request. Current.user and Current.ip_address and Current.user_agent, etc, those feel like "universals." But I've already found myself trying to remember: "for this particular request, does Current.assistant exist or not?" so that makes it feel wrong.

As I'm reflecting on it more, the root of the messiness was that messages to not relate *directly* to assistants — they relate through conversations. But I think it might be a better tradeoff to simply denormalize the tables further, add assistant_id as an attribute of the message model, even though it isn't needed.

That way, new messages are created through `@assistant.messages.create!` and then the side-effect which creates conversations simply plucks it off of `message.assistant_id`

Another small benefit of this is that when you see the route as `/assistants/12345/messages` it seems to imply that messages directly relate to assistants so it's further surprising when you discover they don't have a direct relationship. I'll reflect on all this a bit further before merging this PR in...

Reply:
There might be benefits to attaching messages to assistants directly and detaching conversations. For example, I wonder if there's a situation where I might want to have a conversation with multiple assistants? What if I want to switch between gpt3.5 and 4 mid conversation, or even openai and azure llm models? That's something we can't render right now, as conversations are coupled to assistants. 

Current.assistant feels like it's going to open up a series of counterintuitive situations. I don't think the Current class should even be available outside of controllers. Controller actions reasonably operate within the scope of a request at all times, but Models aren't so clean. We routinely access and trigger model code from background workers, rails console, etc. I can't think of a time when I ran a controller action from within the console or a worker but models are created and destroyed all the time outside of request contexts. What does it mean to bind Current.user or Current.assistant in a background job?